mlir/lib/Conversion/GPUCommon/GPUOpsLowering.cpp

```diff
@@ -359,6 +359,14 @@ GPUFuncOpLowering::matchAndRewrite(gpu::GPUFuncOp gpuFuncOp, OpAdaptor adaptor,
     if (argAttr.empty())
       continue;
 
+    const bool isArgTypeUnchanged =
+        remapping->size == 1 &&
+        llvmFuncOp.getArgument(remapping->inputNo).getType() == argTy;
+    if (isArgTypeUnchanged) {
+      llvmFuncOp.setArgAttrs(remapping->inputNo, argAttr);
+      continue;
+    }
+
```

Comment:
How does this look?

Reply:
I'm not sure "input number" is the right thing to be looking up on the LLVM function? You want the corresponding argument in the output function.

Add a test where the signature is, say, `(memref<16xf32>, !llvm.ptr {llvm.noalias})` to make sure the `noalias` ends up in the right spot.